lib/gis/parser_json.c

```diff
@@ -340,19 +341,30 @@ char *G__json(void)
  */
 void check_create_import_opts(struct Option *opt, char *element, FILE *fp)
 {
-    int i = 0;
+    int i = 0, urlfound;
     int has_import = 0;
     char **tokens;
 
+    G_debug(2,"tokenize opt string: <%s> with '@'", opt->answer);
     tokens = G_tokenize(opt->answer, "@");
     while (tokens[i]) {
         G_chop(tokens[i]);
         i++;
     }
 
+    /* check if tokens[1] contains an URL or name@mapset */
+    G_debug(2,"tokens[1]: <%s>", tokens[1]);
+    if (strstr(tokens[1], "http://") != NULL || strstr(tokens[1], "https://") != NULL || strstr(tokens[1], "ftp://") != NULL ) {
```

Comment:
if you use `strstr() != NULL`, you don't care where the substring you are searching for occurs in `tokens[1]`. Is this intended? Like `strstr("blablablahttp://", "http://")` is not NULL and an url was found.
Why not
```
    if (strncmp(tokens[1], "http://", 7) == 0 || strncmp(tokens[1], "https://", 8) == 0 || strncmp(tokens[1], "ftp://", 6) == 0 ) {
```
?

Reply:
Suggestion happily accepted in d7114ac9a